### PR TITLE
Fix bugs in programmatic cluster config. Add unit test.

### LINKED ISSUE #365: Calling addCluster/User/Context fails if nothing was previously loaded
**Failing Test Code**
```typescript
const kc = new k8s.KubeConfig();
kc.addCluster({
  name: 'test',
  server: `https://localhost:9889`,
  skipTLSVerify: true,
})
///TypeError: Cannot read property 'forEach' of undefined
///        at KubeConfig.addCluster (/Users/test/project/node_modules/@kubernetes/client-node/dist/config.js:176:23)
kc.addUser({
  token: 'test',
  username: 'test',
  name: 'test',
})
kc.addContext({
  cluster: 'test',
  name: 'test',
  user: 'test',
})
kc.setCurrentContext('test')

```
**Code Referenced By StackTrace**
```javascript
// ...
    addCluster(cluster) {
        this.clusters.forEach((c, ix) => {
            if (c.name === cluster.name) {
                throw new Error(`Duplicate cluster: ${c.name}`);
            }
        });
        this.clusters.push(cluster);
    }
    addUser(user) {
        this.users.forEach((c, ix) => {
            if (c.name === user.name) {
                throw new Error(`Duplicate user: ${c.name}`);
            }
        });
        this.users.push(user);
    }
    addContext(ctx) {
        this.contexts.forEach((c, ix) => {
            if (c.name === ctx.name) {
                throw new Error(`Duplicate context: ${c.name}`);
            }
        });
        this.contexts.push(ctx);
    }
// ...
```

It looks like it's as simple as `this.clusters`, `this.users`, `this.contexts` not being defined when the `KubeConfig` is created. The fix would be to just define them when the object is created. 

I would create a PR, but I've been fighting the CLA for a while now, and never got it figured out.

### PR DESCRIPTION
**Pull request by brendandburns**
Fixes #365

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -233,6 +236,9 @@ export class KubeConfig {
     }
 
     public addUser(user: User) {
+        if (!this.users) {
```
Comment (drubin) [inline src/config.ts]: Would it not be better to put this in the constructor? I think it would be more logical and easier to follow too.

Comment (drubin) [reply src/config.ts]: Not a blocker just curious before I add my lgtm (and it merges)

Comment (brendandburns) [reply src/config.ts]: The fields involved are public, so it's always possible that someone will overwrite them with something undefined or null, so the check here is safer.

**Comment by brendandburns** [comment]
Comment addressed, please re-check.

Thanks!

**Comment by drubin** [comment]
/lgtm
Thanks for the clarification

**Pull request merged**